actions/campaigns.go

```diff
@@ -575,3 +577,74 @@ func GetCampaignBounces(c *gin.Context) {
 		"message": "Campaign bounces not found",
 	})
 }
+
+func PatchScheduledCampaign(c *gin.Context) {
+	campaignID, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Id must be an integer",
+		})
+		return
+	}
+
+	campaign, err := storage.GetCampaign(c, campaignID, middleware.GetUser(c).ID)
+	if err != nil {
+		c.JSON(http.StatusNotFound, gin.H{
+			"message": "Campaign not found",
+		})
+		return
+	}
+
+	body := &params.CampaignSchedule{}
+	if err := c.ShouldBind(body); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid parameters, please try again",
+		})
+		return
+	}
+
+	if err := validator.Validate(body); err != nil {
+		c.JSON(http.StatusBadRequest, err)
+		return
+	}
+
+	schAt, err := time.Parse("2006-02-01 15:04:05", body.ScheduledAt)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid parameters, scheduled_at should be format: 2006-02-01 15:04:05",
+		})
+		return
+	}
+
+	var sc *entities.CampaignSchedule
+	sc = &entities.CampaignSchedule{
+		ID:          ksuid.New(),
+		CampaignID:  campaign.ID,
+		ScheduledAt: schAt,
+	}
+
+	// if schedule exist update.
+	if campaign.Schedule != nil {
+		sc = &entities.CampaignSchedule{
+			ID:          campaign.Schedule.ID,
+			CampaignID:  campaignID,
+			ScheduledAt: schAt,
+			CreatedAt:   campaign.Schedule.CreatedAt,
```

Comment:
We don't need to populate CreatedAt and UpdatedAt, gorm handles this for us.

Reply:
@FilipNikolovski I think we need to populate created_at when it's update with the old one

Reply:
Why don't we just update the campaign.Schedule entity here, instead of creating a new one? Also, regarding the CreatedAt, it should remain the same if you perform a `Save` on the entity: https://gorm.io/docs/update.html#Save-All-Fields

Reply:
@FilipNikolovski  Yes you are right. This is done before adding campaign.Schedule field :) I will rework it.

Reply:
@FilipNikolovski it's update if !exist - insert and there i need new object. @FilipNikolovski

Reply:
Yes, but if campaign.Schedule != nil it does exist, so you can just update the scheduled at like so:

```
campaign.Schedule.ScheduledAt = scheduledAt
```
and then pass `campaign.Schedule` to the method. Would something like that work?